lib/slack_rewriter_web/controllers/event_api_controller.ex

```diff
@@ -0,0 +1,56 @@
+defmodule SlackRewriterWeb.EventApiController do
+  use SlackRewriterWeb, :controller
+
+  alias SlackRewriter.Application.UserRegistry
+  alias SlackRewriter.Bridge.Slack
+
+  alias SlackRewriter.Application.SlackRewriter
+
+  def event_handler(
+        conn = %{body_params: %{"challenge" => challenge, "token" => _token, "type" => _type}},
+        _params
+      ) do
+    # recall: verify slack token (better: read doc about signed secrets)
+    conn
+    |> put_resp_content_type("application/x-www-form-urlencoded")
+    |> json(%{challenge: challenge})
+  end
+
+  def event_handler(
+        conn = %{
+          body_params: %{
+            "event" => %{"type" => "message"}
+          }
+        },
+        _params
+      ) do
+    body = conn.body_params
+
+    # istantly respond to slack with 200, OK
+    conn = send_resp(conn, 200, "OK")
+
+    # recall: verify slack token (better: read doc about signed secrets)
+    SlackRewriter.react_to_message(body["event"])
+
+    conn
```

Comment:
sai che non lo so se effettivamente invia la risposta al client e solo dopo far partire il rewriter? non mi sono mai posto la domanda, se hai verificato (con un amichevole sleep ad esempio) / letto che è così allora _today i learned_ 

altrimenti saebbe equivalente a:
```suggestion
    # recall: verify slack token (better: read doc about signed secrets)
    SlackRewriter.react_to_message(body["event"])

    send_resp(conn, 200, "OK")
```
ma hai un bel po' di armi a disposizione per rispondere senza aspettare il rewriter, banalmente spawnare un processo.

Reply:
di base send_resp() è sincrona, quindi prima invia la risposta e poi procede. Tuttavia, in questo caso mi sembra effettivamente più idiomatico usare una spawn()